src/platform/Linux/OTAImageProcessorImpl.cpp

```diff
@@ -169,16 +169,16 @@ void OTAImageProcessorImpl::HandleProcessBlock(intptr_t context)
 
     ByteSpan block   = imageProcessor->mBlock;
     CHIP_ERROR error = imageProcessor->ProcessHeader(block);
-
-    if (error == CHIP_NO_ERROR &&
-        !imageProcessor->mOfs.write(reinterpret_cast<const char *>(block.data()), static_cast<std::streamsize>(block.size())))
+    if (error != CHIP_NO_ERROR)
     {
-        error = CHIP_ERROR_WRITE_FAILED;
+        ChipLogError(SoftwareUpdate, "Image does not contain a valid header");
+        imageProcessor->mDownloader->EndDownload(CHIP_ERROR_INVALID_FILE_IDENTIFIER);
```

Comment:
There can be other errors returned from the parser, too. Shouldn't we rather pass `error` instead of `CHIP_ERROR_INVALID_FILE_IDENTIFIER`?

Reply:
So in `EndDownload`, I map CHIP_ERROR to bdx::StatusCode which the provider will receive. Instead of looking for all possible errors that could occur from parsing the header, I wanted just one value to represent the content is no good, e.g. `bdx::StatusCode::kBadMessageContents`. Alternatively, I could add a bunch of if checks in `EndDownload` but I felt that unnecessary. Let me know if you feel otherwise though.

Reply:
I feel that mapping one error code to another is kind of abuse of the type :), so maybe we should change `EndDownload` to take `bdx::StatusCode` directly instead of `CHIP_ERROR` (if the goal is to send a particular status code), but we can do that later.

Reply:
I'll file an issue to track this for post v1